src/main/java/uk/gov/mca/beacons/service/registrations/GetAllBeaconsService.java

```diff
@@ -25,107 +24,65 @@
 public class GetAllBeaconsService {
 
   private final BeaconRepository beaconRepository;
-  private final BeaconUseRepository beaconUseRepository;
   private final BeaconPersonRepository beaconPersonRepository;
+  private final BeaconUseRepository beaconUseRepository;
+  private final BeaconsRelationshipMapper beaconsRelationshipMapper;
 
   @Autowired
   public GetAllBeaconsService(
     BeaconRepository beaconRepository,
+    BeaconPersonRepository beaconPersonRepository,
     BeaconUseRepository beaconUseRepository,
-    BeaconPersonRepository beaconPersonRepository
+    BeaconsRelationshipMapper beaconsRelationshipMapper
   ) {
     this.beaconRepository = beaconRepository;
     this.beaconPersonRepository = beaconPersonRepository;
     this.beaconUseRepository = beaconUseRepository;
+    this.beaconsRelationshipMapper = beaconsRelationshipMapper;
   }
 
   public List<Beacon> findAll() {
     final List<Beacon> allBeacons = beaconRepository.findAll();
+    final List<BeaconPerson> allBeaconPersons = beaconPersonRepository.findAll();
+    final List<BeaconUse> allBeaconUses = beaconUseRepository.findAll();
     if (allBeacons.size() == 0) return emptyList();
 
-    final Map<UUID, List<BeaconUse>> usesGroupedByBeaconId = getAllUsesGroupedByBeaconId();
-    final Map<PersonType, List<BeaconPerson>> personsGroupedByType = getAllPersonsGroupedByType();
-    final Map<UUID, BeaconPerson> ownersGroupedByBeaconId = getAllOwnersGroupedByBeaconId(
-      personsGroupedByType
-    );
-    final Map<UUID, List<BeaconPerson>> emergencyContactsGroupedByBeaconId = getAllContactsGroupedByBeaconId(
-      personsGroupedByType
-    );
-
-    return mapBeaconRelationships(
+    return beaconsRelationshipMapper.getMappedBeacons(
       allBeacons,
-      usesGroupedByBeaconId,
-      ownersGroupedByBeaconId,
-      emergencyContactsGroupedByBeaconId
+      allBeaconPersons,
+      allBeaconUses
     );
   }
 
-  private List<Beacon> mapBeaconRelationships(
-    final List<Beacon> allBeacons,
-    final Map<UUID, List<BeaconUse>> usesGroupedByBeaconId,
-    final Map<UUID, BeaconPerson> ownersGroupedByBeaconId,
-    final Map<UUID, List<BeaconPerson>> emergencyContactsGroupedByBeaconId
-  ) {
-    List<Beacon> mappedBeacons = new ArrayList<Beacon>();
+  public BeaconDTO find(UUID id) {
+    final Optional<Beacon> beacon = beaconRepository.findById(id);
+    final List<BeaconPerson> allBeaconPersons = beaconPersonRepository.findAll();
+    final List<BeaconUse> allBeaconUses = beaconUseRepository.findAll();
```

Comment:
As discussed, you can write custom queries here on the repositories to extract all beacon persons for a given id and all uses for a given id.